src/main/native/windows/file.cc

```diff
@@ -781,5 +781,46 @@ bool GetCwd(std::wstring* result, DWORD* err_code) {
   }
 }
 
+std::wstring GetCorrectCasing(const std::wstring& abs_path, bool with_unc) {
+  if (!HasDriveSpecifierPrefix(abs_path.c_str())) {
+    return L"";
+  }
+  std::wstring path = Normalize(abs_path);
+  std::unique_ptr<wchar_t[]> result(new wchar_t[4 + path.size() + 1]);
+  wcscpy(result.get(), L"\\\\?\\");
+  wcscpy(result.get() + 4, path.c_str());
+  result[4] = towupper(result[4]);
+  wchar_t* start = result.get() + 7;
```

Comment:
There is a lot of implicit knowledge in this code which makes it hard to read, e.g. where does the 7 come from?

Reply:
is this from UNC (if so make it conditional)?

Reply:
Added comment.